source/sixve.f90

```diff
@@ -131,92 +131,63 @@ subroutine sumpos
-!    &jm,k,kpz,kzz,l,lkk,ll,m,mkk,napxto,ncorruo,ncrr,nd,nd2,ndafi2,    &
-!    &nerror,nlino,nlinoo,nmz,nthinerr
-!     double precision alf0s1,alf0s2,alf0s3,alf0x2,alf0x3,alf0z2,alf0z3,&
-!    &amp00,bet0s1,bet0s2,bet0s3,bet0x2,bet0x3,bet0z2,bet0z3,chi,coc,   &
-!    &dam1,dchi,ddp1,dp0,dp00,dp10,dpoff,dpsic,dps0,dsign,gam0s1,gam0s2,&
-!    &gam0s3,gam0x1,gam0x2,gam0x3,gam0z1,gam0z2,gam0z3,phag,r0,r0a,rat0,&
-!    &rdev,rmean,rsqsum,rsum,sic,tasia56,tasiar16,tasiar26,tasiar36,    &
-!    &tasiar46,tasiar56,tasiar61,tasiar62,tasiar63,tasiar64,tasiar65,   &
-!    &taus,x11,x13
-
-save
+  real(kind=fPrec) dpoff, hv(6,2,nblo)
 
 #ifdef FLUKA
-!     A.Mereghetti and D.Sinuela Pastor, for the FLUKA Team
-!     last modified: 11-06-2014
-!     entirely re-initialise to 0.0 hv(...) and bl1v(...) arrays
-!     inserted in main code by the 'fluka' compilation flag
-  do ia=1,npart
-    do k=1,nblo
+  ! Entirely re-initialise to 0.0 hv(...) and bl1v(...) arrays
+  hv(:,:,:)   = zero
+  bl1v(:,:,:) = zero
+#endif
+  do ia=1,napx
+    do k=1,mblo
+      jm  = mel(k)
+      ikk = mtyp(k,1)
       do lkk=1,2
         do mkk=1,6
-          hv(mkk,lkk,ia,k)=zero
-          bl1v(mkk,lkk,ia,k)=zero
-        end do
-      end do
-    end do
-  end do
-#endif
-  do 440 k=1,mblo
-    jm=mel(k)
-    ikk=mtyp(k,1)
-    do lkk=1,2
-      do mkk=1,6
-        do ia=1,napx
-          dpoff=dpsv(ia)*c1e3
-          if(abs(dpoff).le.pieni) dpoff=one
-          hv(mkk,lkk,ia,1)=al(mkk,lkk,ia,ikk)
-          if(mkk.eq.5.or.mkk.eq.6) then
-            hv(mkk,lkk,ia,1)=hv(mkk,lkk,ia,1)/dpoff
+          dpoff = dpsv(ia)*c1e3
+          if(abs(dpoff) <= pieni) dpoff = one
+          hv(mkk,lkk,1) = al(mkk,lkk,ia,ikk)
+          if(mkk == 5 .or. mkk == 6) then
+            hv(mkk,lkk,1) = hv(mkk,lkk,1)/dpoff
           end if
         end do
       end do
-    end do
-    if(jm.eq.1) goto 410
-    do j=2,jm
-      ikk=mtyp(k,j)
-      do lkk=1,2
-        do ia=1,napx
-          dpoff=dpsv(ia)*c1e3
-          if(abs(dpoff).le.pieni) dpoff=one
-          hv(1,lkk,ia,j)=hv(1,lkk,ia,j-1)*al(1,lkk,ia,ikk)+ hv(3,lkk,ia,j-1)*al(2,lkk,ia,ikk)
-          hv(2,lkk,ia,j)=hv(2,lkk,ia,j-1)*al(1,lkk,ia,ikk)+ hv(4,lkk,ia,j-1)*al(2,lkk,ia,ikk)
-          hv(3,lkk,ia,j)=hv(1,lkk,ia,j-1)*al(3,lkk,ia,ikk)+ hv(3,lkk,ia,j-1)*al(4,lkk,ia,ikk)
-          hv(4,lkk,ia,j)=hv(2,lkk,ia,j-1)*al(3,lkk,ia,ikk)+ hv(4,lkk,ia,j-1)*al(4,lkk,ia,ikk)
-!hr05         hv(5,lkk,ia,j)=hv(5,lkk,ia,j-1)*al(1,lkk,ia,ikk)+ hv(6,   &
-!hr05&lkk,ia,j-1)*al(2,lkk,ia,ikk)+al(5,lkk,ia,ikk)/dpoff
-          hv(5,lkk,ia,j)=(hv(5,lkk,ia,j-1)*al(1,lkk,ia,ikk)+ hv(6,lkk,ia,j-1)*al(2,lkk,ia,ikk))+al(5,lkk,ia,ikk)/dpoff
-!hr05         hv(6,lkk,ia,j)=hv(5,lkk,ia,j-1)*al(3,lkk,ia,ikk)+ hv(6,   &
-!hr05&lkk,ia,j-1)*al(4,lkk,ia,ikk)+al(6,lkk,ia,ikk)/dpoff
-          hv(6,lkk,ia,j)=(hv(5,lkk,ia,j-1)*al(3,lkk,ia,ikk)+ hv(6,lkk,ia,j-1)*al(4,lkk,ia,ikk))+al(6,lkk,ia,ikk)/dpoff
+      if(jm > 1) then
```

Comment:
Logic used to be jm != 1. Are you sure it (eg. `mel`) cannot be 0 or negative?

Reply:
The loop runs from 2 to jm, so even if jm is 0, there is nothing to loop over.